### PR TITLE
Add ETH/CTDL and BTC/CTDL chainlink provider

### PR DESCRIPTION
**Pull request by shuklaayush**
- Uses BTC-ETH, WBTC-BTC and CTDL curve v2 pool's `price_oracle` to get ETH-CTDL price feed
- Uses BTC-WBTC and CTDL curve v2 pool's `price_oracle` to get BTC-CTDL price feed

Addresses #140

### CODE REVIEW

File: foundry.toml [hunk 1]
```diff
@@ -8,5 +8,6 @@ remappings = [
     'openzeppelin-contracts/=lib/openzeppelin-contracts/contracts/',
     'forge-std/=lib/forge-std/src/',
 ]
+verbosity = 3
```
Comment (dapp-whisperer) [inline foundry.toml:11]: what's the reason behind this change?

File: src/oracles/CtdlBtcChainlinkProvider.sol [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+    ICurveCryptoSwap public immutable ctdlWbtcCurvePool;
+
+    // Price feeds
+    IAggregatorV3Interface public immutable wbtcBtcPriceFeed;
+
+    /// =====================
+    /// ===== Constants =====
+    /// =====================
+
+    uint256 constant PRECISION = 10**18;
```
Comment (dapp-whisperer) [inline src/oracles/CtdlBtcChainlinkProvider.sol]: to confirm precision never used in this variant?

**Review by dapp-whisperer — APPROVED** [review]
lgtm, needs some tests so please create a separate issue for that

File: foundry.toml (hunk 1 above)
Comment (shuklaayush) [reply foundry.toml:11]: This is to print the trace for failed tests by default. Essentially, to avoid running tests with `-vvv` every time

File: src/oracles/CtdlBtcChainlinkProvider.sol (hunk 2 above)
Comment (shuklaayush) [reply src/oracles/CtdlBtcChainlinkProvider.sol]: Yep, I thought I removed this. Good catch
